pkg/apis/mongodb/v1/mongodb_types.go

```diff
@@ -58,15 +58,23 @@ type TLS struct {
 	// +optional
 	Optional bool `json:"optional"`
 
-	// ServerSecretName is the name of a secret containing a private key and certificate to use for TLS
+	// CertificateKeySecret is a reference to a Secret containing a private key and certificate to use for TLS
 	// The key and cert are expected to be PEM encoded and available at "tls.key" and "tls.crt"
 	// +optional
-	ServerSecretName string `json:"serverSecretName"`
+	CertificateKeySecret LocalObjectReference `json:"certificateKeySecretRef"`
 
-	// CAConfigMapName is the name of a ConfigMap containing the certificate for the CA which signed the server certificates
+	// CaConfigMap is a reference to a ConfigMap containing the certificate for the CA which signed the server certificates
 	// The certificate is expected to be available under the key "ca.crt"
 	// +optional
-	CAConfigMapName string `json:"caConfigMapName"`
+	CaConfigMap LocalObjectReference `json:"caConfigMapRef"`
+}
+
+// LocalObjectReference is a reference to another Kubernetes object by name.
+// TODO: Replace with a type from the K8s API. CoreV1 has an equivalent
```

Comment:
I assume then that this TODO won't show in our CRD, is that right?

Reply:
Exactly! There TODO is on the `Name` field which the Operator SDK uses when generating the CRD.